pkg/cmd/codespace/rebuild.go

```diff
@@ -51,11 +52,29 @@ func (a *App) Rebuild(ctx context.Context, codespaceName string, full bool) (err
 	}
 	defer safeClose(session, &err)
 
-	err = session.RebuildContainer(ctx, full)
```

Comment:
Wild idea: instead of moving RebuildContainer off of the "session" object, instantiate a session object that dispatches some operations to Liveshare and some to gRPC. Over time, more and more operations would go to gRPC than Liveshare, but callers of these methods would be none the wiser. That way codespace commands do not have to know about the session internals. What do you think?

Reply:
:+1: this is something we [discussed a bit](https://github.com/cli/cli/pull/6358#discussion_r982596498) back in the original gRPC PR. The idea at that time was to make an interface like `RPCInvoker` where `liveshare.Session` implemented as well as `grpc.Client`. 

In the interim migration we could have a third implementation that took both types of clients in the constructor and selected one or the other based on the method and slowly move all of the implementations to use `grpc.Client`. Then at the end of migration we just get rid of that wrapper type and start using `grpc.Client` for everything and the code is none the wiser.

Reply:
@cmbrose Agreed with that approach. I do feel that the `err = session.RebuildContainer(ctx, full)` call point should not change like it currently does in this PR; only the implementation of `session.RebuildContainer` should change.

Reply:
The issue with changing the implementation in `session` is that it's very Live Share-specific and I'm not sure if we plan to keep it around with the switch to our new connection stack in the coming months. 

If I'm understanding correctly, I think Caleb's idea is to replace all of the RPC calls currently being done with a new object that calls the actual implementation we want (either Live Share RPC via the `session` object or gRPC). So something like `rpc.RebuildContainer(ctx, full)` would be the new signature and the caller would be none the wiser of what's going on under the hood`.

Reply:
PR to make this change: https://github.com/cli/cli/pull/6657